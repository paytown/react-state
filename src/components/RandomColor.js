import React, { PureComponent } from 'react';

export default class RandomColor extends PureComponent {
  state={
    color: '',
    trigger: new Date()
  };

  randomColor = () => {
    const colors = ['#CFF09E', '#A8DBA8', '#79BD9A', '#3B8686', '#0B486B'];
    const n = Math.floor(Math.random() * colors.length);
    return colors[n];
  }

  componentDidMount() {
    this.intervalId = setInterval(() => {
      this.setState({ color: this.randomColor(), trigger: new Date() });
    }, 2000);
  }

  componentDidUpdate(prevProps, prevState) {
    if(prevState.color === this.state.color) {
      this.setState({ color: '#FF4E50' });
    }
  }

  componentWillUnmount() {
    this.intervalId && clearInterval(this.intervalId);
  }

  render() {
    const { color } = this.state;
    const styles = {
      background: color,
      width: '100%',
      height: '100vh',
      transition: 'background-color 2500ms linear'
    };
    return <div style={styles}></div>;
  }
}
