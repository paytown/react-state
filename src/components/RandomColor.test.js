import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('coloooor', () => {
  it('colors fade randomly', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
