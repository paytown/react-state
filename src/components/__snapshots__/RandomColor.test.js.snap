// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coloooor colors fade randomly 1`] = `
<div
  style={
    Object {
      "background": "",
      "height": "100vh",
      "transition": "background-color 2500ms linear",
      "width": "100%",
    }
  }
/>
`;
